# Kinematic configuration for the MIT cheetah 
# Initial pose
# initial postion: x,y,z
# initial orientation: x,y,z,w
# Links 
# base_link: base_link frame
# Joints 
# joint_names: joint names
# joint_actuator_names: joint actuator names
# init_joint_positions: initial joint positions 
# init_joint_torques: initial joints torques 
# joint_velocities: initial joint velocities
# stance_joint_positions: starting standing configuration joint positions 
# stance_joint_velocities: starting standing configuration joint velocities

initial_pose:
  postion: [0.0, 0.0, 0.06] 
  orientation: [0.0, 0.0, 0.0, 1.0]

links:
  base_link: "trunk"

joints:
  num_joints: 12

  joint_names: ["RL_hip_joint", "FL_hip_joint", "RR_hip_joint", "FR_hip_joint",
                "RL_thigh_joint", "FL_thigh_joint", "RR_thigh_joint", "FR_thigh_joint",
                "RL_calf_joint", "FL_calf_joint", "RR_calf_joint", "FR_calf_joint"]

  joint_actuator_names: ["RL_hip_motor", "RL_thigh_motor", "RL_calf_motor", 
                         "FL_hip_motor", "FL_thigh_motor", "FL_calf_motor",
                         "RR_hip_motor", "RR_thigh_motor", "RR_calf_motor",
                         "FR_hip_motor", "FR_thigh_motor", "FR_calf_motor"]

  # Maps to joint names
  init_joint_positions: [0.0, 0.0, 0.0, 0.0, 1.61, 1.61, 1.61, 1.61, -2.78, -2.78, -2.78, -2.78]
  
  # Maps to joint actuator names 
  init_joint_torques: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]

# s_diagonal: diagonal weights on least squares (Ax-b)*S*(Ax-b)
# w_diagonal: diagonal weight on forces in fT*W*f in format f = [RL, FL, RR, FR] (fx,fy,fz)
# kff: feed forward gain on [vx_d, vy_d, m*g, roll_dot, pitch_dot, yaw_dot]
# kp_p: COM position Kp
# kp_w: COM orientation Kp
# kd_p: COM linear velocity Kd
# kd_w: COM angular velocity Kd
control:
  frequency: 100.0 # Hz
  torque_min: -20.0
  torque_max: 20.0

  # Standing 
  s_diagonal: [1.0, 1.0, 1.0, 10.0, 10.0, 5.0]
  w_diagonal: [0.00001]
  
  kff: [0.0, 0.0, 0.20, 0.0, 0.0, 0.0]
  kp_p: [100.0, 100.0, 150.0]
  kp_w: [3000.0, 3000.0, 3000.0]

  kd_p: [50.0, 50.0, 50.0]
  kd_w: [250.0, 250.0, 250.0]
